File: MicroBenchmarks/Fib32-and-Fact20.py
from time import time

def started_time():
	return int(1000 * time())
def elapsed_time(start):
	return int(1000 * time()) - start
def Fib(n):
    if n < 2:
        return n
    return Fib(n - 1) + Fib(n - 2)
def Fact(n):
    if 1 < n:
        return n * Fact(n - 1)
    return 1

print("Python... About to compute Fib(32) (only once)...")
print("(Press return)")
input()

t0 = started_time()
fib32 = Fib(32)
elapsed = elapsed_time(t0)
print("Fib(32) = " + str(fib32) + " in " + str(elapsed) + " ms")

print()
print("Python... About to compute 20! x 1,000,000 times...")
print("(Press return)")
input()

t0 = started_time()
for i in range(0, 1000000):
	fact20 = Fact(20)
elapsed = elapsed_time(t0)
print("20! = " + str(fact20) + " x 1,000,000 times in " + str(elapsed) + " ms")
print("(Press return to exit)")
input()

File: MicroBenchmarks/Fib32-and-Fact20.js
function Fib(n) {
  return n < 2 ? n : Fib(n - 1) + Fib(n - 2);
}

function Fact(n) {
  return 1 < n ? n * Fact(n - 1) : 1;
}

var fact20, fib32, t0, elapsed;

WScript.StdOut.WriteLine("MS JScript... About to compute Fib(32) (only once)...");
WScript.StdOut.WriteLine();

t0 = new Date().valueOf();
fib32 = Fib(32);
elapsed = new Date().valueOf() - t0;

WScript.StdOut.WriteLine("Fib(32) = " + fib32.toString() + " in " + elapsed.toString() + " ms");
WScript.StdOut.WriteLine();

WScript.StdOut.WriteLine("MS JScript... About to compute 20! x 1,000,000 times...");
WScript.StdOut.WriteLine();

t0 = new Date().valueOf();
for (var i = 0; i < 1000000; i++)
{
  fact20 = Fact(20);
}
elapsed = new Date().valueOf() - t0;

WScript.StdOut.WriteLine("20! = " + fact20.toString() + " x 1,000,000 times in " + elapsed.toString() + " ms");
WScript.StdOut.WriteLine();
